### ISSUE TITLE
Bug - yourRate is sometimes parsed incorrectly

### ISSUE THREAD

**Issue by coffee-converter** [grepole121/quickswap_apy#1]
https://github.com/grepole121/quickswap_apy/blob/5524cb5e4a09e78c16b8327e56661dbe9db8f5ae/displayAPYs.js#L74

Since this particular element may contain the extension-injected dollar amount in parentheses, the regex isn't enough to parse the quick amount, and so it can end up including some of the digits from the dollar amount:

<img width="1258" alt="Screen Shot 2021-05-04 at 12 17 11 PM" src="https://user-images.githubusercontent.com/61335667/117043312-ad90f500-acd2-11eb-9fac-fffdb17b5b87.png">

**Comment by grepole121** [comment]
Thanks for bringing this up. 
I have removed the regex and changed it to use a substring ignoring the first character (the emoji). parseFloat just ignores the later data after the space.
Did you actually notice this causing any issues with the extension or just in the code because I see the problem here but this shouldn't have caused any major problems as it's after multiple decimal places so the effect would be very small?

**Comment by grepole121** [comment]
Just out of curiosity, what IDE are you using here and how are you doing the debugging for the script as it would need to run in a browser wouldn't it? I mostly just used console.log() for my debugging as I don't have much experience with JavaScript and learned to build this extension

**Comment by coffee-converter** [comment]
> Thanks for bringing this up.
> I have removed the regex and changed it to use a substring ignoring the first character (the emoji). parseFloat just ignores the later data after the space.
> Did you actually notice this causing any issues with the extension or just in the code because I see the problem here but this shouldn't have caused any major problems as it's after multiple decimal places so the effect would be very small?

Yeah the effect should be very small...  I actually started looking into it because I noticed a large discrepancy in my "your deposits in this pool" number on a pool after a long while of having the page open and then clicking the extension to refresh...it would be a few percent too high...reloading the page and clicking the extension again corrected the value.  I'm still trying to isolate that bug.

**Comment by coffee-converter** [comment]
> Just out of curiosity, what IDE are you using here and how are you doing the debugging for the script as it would need to run in a browser wouldn't it? I mostly just used console.log() for my debugging as I don't have much experience with JavaScript and learned to build this extension

I was just using the Chrome dev tools (Sources tab) debugger...much handier than adding console statements!

**Comment by coffee-converter** [comment]
> > Thanks for bringing this up.
> > I have removed the regex and changed it to use a substring ignoring the first character (the emoji). parseFloat just ignores the later data after the space.
> > Did you actually notice this causing any issues with the extension or just in the code because I see the problem here but this shouldn't have caused any major problems as it's after multiple decimal places so the effect would be very small?
> 
> Yeah the effect should be very small... I actually started looking into it because I noticed a large discrepancy in my "your deposits in this pool" number on a pool after a long while of having the page open and then clicking the extension to refresh...it would be a few percent too high...reloading the page and clicking the extension again corrected the value. I'm still trying to isolate that bug.

@grepole121 I believe the main issue I noticed with occasional wrong deposit amounts is because once the extension has overwritten the textContent of the "your rate" line (to include the dollar amount), the UI no longer updates that cell with new rate info.  Maybe append/update a new div with the dollar amount instead?  Might fix it.

**Comment by grepole121** [comment]
So QuickSwap doesn't update "Your rate" part of the website after the script runs and puts the dollar equivalent in brackets?

**Comment by coffee-converter** [comment]
> So QuickSwap doesn't update "Your rate" part of the website after the script runs and puts the dollar equivalent in brackets?

Yeah, I believe that's what's happening.  I didn't test it too much but that's what it looked like to me.  On one tab I ran the extension once and then let it sit there as Quickswap updated values.  On another tab I opened the same page but didn't trigger the extension, and the rate values there updated while the previous tab's rate values remained the same.
